OpenRA.Mods.Common/Traits/AttackMove.cs

```diff
@@ -123,7 +123,7 @@ protected virtual IEnumerable<Order> OrderInner(World world, CPos cell, MouseInp
 
 		public override void SelectionChanged(World world, IEnumerable<Actor> selected)
 		{
-			subjects = selected.SelectMany(a => a.TraitsImplementing<AttackMove>()
+			subjects = selected.Where(s => !s.IsDead).SelectMany(a => a.TraitsImplementing<AttackMove>()
```

Comment:
@teinarss (or anyone else that knows), how does this àdded `Where` stack up in terms of perf vs. just putting that check as first check in the `SelectMany`?

Reply:
I guess performance wise it is better to sort out the dead before doing a trait lookup (which crashes the game) plus I am not sure if it is syntactically possible to do a `SelectMany` which includes a binary check.

Reply:
Something like
```
subjects = selected.SelectMany(a =>
			{
				if (!a.IsDead)
					return a.TraitsImplementing<AttackMove>().Select(am => new TraitPair<AttackMove>(a, am));

				return Enumerable.Empty<TraitPair<AttackMove>>();
			}).ToArray();
```
should work, but I don't think it is really worth it here.